### DISCUSSION TITLE
Running pretix and JS frontend on the same VPS?

### DISCUSSION THREAD

**Discussion by carl-egge** [pretix/pretix#3406] [Q&A] [answered]
Hej guys,

I followed the [small-scale installation guide](https://docs.pretix.eu/en/latest/admin/installation/manual_smallscale.html) and deployed the pretix application on a cloud server. I was wondering what is the best way to also run a JavaScript frontend on the same server. My desired behavior would be that all routes like "/" or "/about" are handled by my frontend and that the ticket selling works with pretix and runs under "/shop" or something. I know that this not a functionality that pretix is supposed to provide, but I figure it is a typical use case for some developers and I just wanted to start a discussion on what is the best way to do this.

The obvious way is to configure nginx as a reverse proxy and let it handle the routing, but I failed to get this to work until now. pretix seems to send only 404 pages if it doesn't run on the "/" home route. Maybe someone can give me a hint for a configuration.

I also thought it would be a possibility to run pretix as the docker version and then spin up another docker container for the frontend, but I wouldn't know how to do this since I am not experienced with docker-compose.

What do you guys think? Can someone maybe help me with this?

Best regards, Carl

**Comment by raphaelm** [comment, marked as answer]
pretix can not be run under a subpath. You can of course run it on a different subdomain.

**Reply by carl-egge** [reply]
Thank you for the quick reply!

That's exactly what I am doing now. I have my frontend on the main domain and pretix running under a subdomain. I also plan on embedding the pretix widget in the frontend, which then only forwards to the subdomain pretix page for the checkout pages. This should be possible, right?

Good to know that I am not just too stupid to run it under a subpath.

One more question, if I may, Since I installed pretix manually is there any way for me to influence the look of the Welcome page (the one that says: "Hello! This is a self-hosted installation of pretix, the friendly open source ticket store. . . ") ?

**Reply by raphaelm** [reply]
You could do it in nginx with a
```
        location = / {
            return 301 https://somewhere;
        }
```
block

**Reply by carl-egge** [reply]
Thank you for the help!
